typing/typedtree.mli

```diff
@@ -18,11 +24,21 @@ open Types
 (* Value expressions for the core language *)
 
 type partial = Partial | Total
+
 type optional = Required | Optional
+(** [optional] is used to pass knowledge of optional arguments up to
+    [Translcore.transl_apply].
+    When introduced in 2000, this enabled a more efficient code generation for
```

Comment:
This was used as an abstraction for labels, but since we are now keeping the labels in the typed tree, this is redundant.